fig.go

```diff
@@ -112,14 +113,15 @@ func defaultFig() *fig {
 }
 
 type fig struct {
-	filename   string
-	dirs       []string
-	tag        string
-	timeLayout string
-	useEnv     bool
-	useStrict  bool
-	ignoreFile bool
-	envPrefix  string
+	filename        string
+	dirs            []string
+	tag             string
+	timeLayout      string
+	useEnv          bool
+	useConstantCase bool
```

Comment:
We can introduce an abstraction here so that multiple strategies can coexist without cluttering this struct and `formatEnvKey` with each option:

```go

func defaultFig() *fig {
	return &fig{
		envKeyTransform: strings.ToUpper
	}
}

type fig struct {
    envKeyTransform func(string) string
}

...

func (f *fig) formatEnvKey(key string) string {
   ..
   key = f.envKeyTransform(key)
}

...

func UseConstantCaseForEnvKeys() Option {
	return func(f *fig) {
		f. envKeyTransform = toConstantCase
	}
}


```

Reply:
Should we pass the env prefix to transformers? it's needed for adding prefix to the generated env key.

Reply:
Yeah we should add it